src/emulator/gui/src/game_selector.cpp

```diff
@@ -179,22 +190,105 @@ void draw_game_selector(GuiState &gui, HostState &host) {
                     }
                     if (ImGui::IsMouseClicked(0))
                         selected[0] = true;
+                    if (gui.title_id != game.title_id)
+                        gui.title_id = game.title_id;
+                }
+
+                const auto games_index = std::find_if(gui.game_selector.games.begin(), gui.game_selector.games.end(), [&game](const Game &g) {
+                    return g.app_ver == game.app_ver && g.title == game.title && g.title_id == game.title_id;
+                });
+
+                if (!fs::exists(game_path / game.title_id)) {
+                    gui.game_selector.games.erase(games_index);
+                    if (gui.delete_game_popup)
+                        gui.delete_game_popup = false;
+                }
+
+                if (gui.title_id == game.title_id) {
+
+                    if (ImGui::BeginPopupContextItem("#game_context_menu")) {
+                        if (ImGui::MenuItem("Boot", game.title.c_str()))
+                            selected[0] = true;
+                        if (ImGui::BeginMenu("Copy game info")) {
+                            if (ImGui::MenuItem("ID and Name")) {
+                                ImGui::LogToClipboard();
+                                ImGui::LogText("%s [%s]", game.title_id.c_str(), game.title.c_str());
+                                ImGui::LogFinish();
+                            }
+                            if (ImGui::MenuItem("ID")) {
+                                ImGui::LogToClipboard();
+                                ImGui::LogText("%s", game.title_id.c_str());
+                                ImGui::LogFinish();
+                            }
+                            if (ImGui::MenuItem("Name")) {
+                                ImGui::LogToClipboard();
+                                ImGui::LogText("%s", game.title.c_str());
+                                ImGui::LogFinish();
+                            }
+                            ImGui::EndMenu();
+                        }
+                        if (ImGui::BeginMenu("Remove")) {
+                            if (ImGui::MenuItem("Game"))
+                                gui.delete_game_popup = true;
+                            if (ImGui::MenuItem("Shader Log")) {
+                                if (fs::exists(shaderlog_path / game.title_id))
+                                    fs::remove_all(shaderlog_path / game.title_id);
+                            }
+                            ImGui::EndMenu();
+                        }
+                        ImGui::EndPopup();
+                    }
+
+                    if (gui.delete_game_popup)
```

Comment:
I feel like this if statement should cover more. Actually, move this entire delete dialog to it's own function! `draw_game_selector` is getting cramped.

Reply:
? I can't here this popup need run inside for and inside check titille id

Reply:
Pass all the info you need to your function :)

It will make everything look cleaner. When I was doing icon work I had trouble getting around this function, so I would prefer we work to clean it up.

Reply:
i have no understund ? like i said, this code no can works outside loop 🤔

Reply:
Throw this code block in a function and call that function in the loop

Reply:
Meaning create a function named draw_game_info_popup where popup stuff happens, and call this from thr loop

Reply:
i have no uderstund what you want said exactlu, like i said, this function no can works outside loop 🤔

Reply:
```cpp
void draw_game_info_popup_dialog(GuiState &gui) {
                if (ImGui::BeginPopupContextItem("#game_context_menu")) {
                        if (ImGui::MenuItem("Boot", game.title.c_str()))
                            selected[0] = true;
                        if (ImGui::BeginMenu("Copy game info")) {
                            if (ImGui::MenuItem("ID and Name")) {
                                ImGui::LogToClipboard();
                                ImGui::LogText("%s [%s]", game.title_id.c_str(), game.title.c_str());
                                ImGui::LogFinish();
                            }
                            if (ImGui::MenuItem("ID")) {
                                ImGui::LogToClipboard();
                                ImGui::LogText("%s", game.title_id.c_str());
                                ImGui::LogFinish();
                            }
                            if (ImGui::MenuItem("Name")) {
                                ImGui::LogToClipboard();
                                ImGui::LogText("%s", game.title.c_str());
                                ImGui::LogFinish();
                            }
                            ImGui::EndMenu();
                        }
                        if (ImGui::BeginMenu("Remove")) {
                            if (ImGui::MenuItem("Game"))
                                gui.delete_game_popup = true;
                            if (ImGui::MenuItem("Shader Log")) {
                                if (fs::exists(shaderlog_path / game.title_id))
                                    fs::remove_all(shaderlog_path / game.title_id);
                            }
                            ImGui::EndMenu();
                        }
                        ImGui::EndPopup();
                    }
}
```

Reply:
Do so with delete one

Reply:
you with i create news file just for that ?

Reply:
New functions

Reply:
ah you want in the top file ?

Reply:
but so yes, exactly lik i have thinks, rip, no can works, obligatory need run it inside loop here for all game.title etc variable news function don't have look with for game Games

Reply:
You can separate basically any piece of code into a function. Give us the error you're getting in #development and we'll help you solve it.